app/models/outbound_webhook.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'faraday'
+
+class OutboundWebhook < ActiveRecord::Base
+  has_soft_deletion default_scope: true
+
+  belongs_to :project
+  belongs_to :stage
+
+  validates :url, uniqueness: {
+    scope: :stage_id,
+    conditions: -> { where("deleted_at IS NULL") },
```

Comment:
I don't think the default scope works in the validator.  Tests broke if I removed it.